sdk/eventhub/Azure.Messaging.EventHubs/tests/Primitives/EventProcessorTests.cs

```diff
@@ -22,6 +24,947 @@ namespace Azure.Messaging.EventHubs.Tests
     [TestFixture]
     public class EventProcessorTests
     {
+        /// <summary>
+        ///   Provides test cases for the constructor tests.
+        /// </summary>
+        ///
+        public static IEnumerable<object[]> ConstructorCreatesDefaultOptionsCases()
+        {
+            var connectionString = "Endpoint=sb://somehost.com;SharedAccessKeyName=ABC;SharedAccessKey=123;EntityPath=somehub";
+            var credential = Mock.Of<TokenCredential>();
+
+            yield return new object[] { new BasicProcessorMock(99, "consumerGroup", connectionString), "connection string with default options" };
+            yield return new object[] { new BasicProcessorMock(99, "consumerGroup", connectionString, default(EventProcessorOptions)), "connection string with explicit null options" };
+            yield return new object[] { new BasicProcessorMock(99, "consumerGroup", "namespace", "hub", credential), "namespace with default options" };
+            yield return new object[] { new BasicProcessorMock(99, "consumerGroup", "namespace", "hub", credential, default(EventProcessorOptions)), "namespace with explicit null options" };
+        }
+
+        /// <summary>
+        ///   Verifies functionality of the <see cref="EventProcessor{TPartition}" />
+        ///   constructor.
+        /// </summary>
+        ///
+        [Test]
+        [TestCase(-100)]
+        [TestCase(-10)]
+        [TestCase(-1)]
+        [TestCase(0)]
+        public void ConstructorValidatesTheConsumerGroup(int batchSize)
```

Comment:
nit: method name is not accurate. `ConstructorValidatesTheEventBatchMaximumCount`.